src/main/java/uk/ac/ebi/eva/pipeline/io/mappers/VariantVcfFactory.java

```diff
@@ -243,82 +242,22 @@ protected VariantKeyFields createVariantsFromIndelNoEmptyRefAlt(int position, St
-                        }
-                        genotypeStr.append(genotype.isPhased() ? "|" : "/");
-                    }
-                    sampleField = genotypeStr.substring(0, genotypeStr.length()-1);
-                        
-                } else if (formatField.equalsIgnoreCase("GL")
-                        || formatField.equalsIgnoreCase("PL")
-                        || formatField.equalsIgnoreCase("GP")) {
-                    // All-alleles present and not haploid
-                    if (!sampleField.equals(".") && genotype != null
-                            && (genotype.getCode() == AllelesCode.ALLELES_OK
-                            || genotype.getCode() == AllelesCode.MULTIPLE_ALTERNATES)) {
-                        String[] likelihoods = sampleField.split(",");
-
-                        // If only 3 likelihoods are represented, no transformation is needed
-                        if (likelihoods.length > 3) {
-                            // Get alleles index to work with: if both are the same alternate,
-                            // the combinations must be run with the reference allele.
-                            // Otherwise all GL reported would be alt/alt.
-                            int allele1 = genotype.getAllele(0);
-                            int allele2 = genotype.getAllele(1);
-                            if (genotype.getAllele(0) == genotype.getAllele(1) && genotype.getAllele(0) > 0) {
-                                allele1 = 0;
-                            }
-
-                            // If the number of values is not enough for this GT
-                            int maxAllele = allele1 >= allele2 ? allele1 : allele2;
-                            int numValues = (int) (((float) maxAllele * (maxAllele + 1)) / 2) + maxAllele;
-                            if (likelihoods.length < numValues) {
-                                throw new NonStandardCompliantSampleField(
-                                    formatField, sampleField, String.format("It must contain %d values", numValues));
-                            }
-
-                            // Genotype likelihood must be distributed following similar criteria as genotypes
-                            String[] alleleLikelihoods = new String[3];
-                            alleleLikelihoods[0] = likelihoods[(int) (((float) allele1 * (allele1 + 1)) / 2) + allele1];
-                            alleleLikelihoods[1] = likelihoods[(int) (((float) allele2 * (allele2 + 1)) / 2) + allele1];
-                            alleleLikelihoods[2] = likelihoods[(int) (((float) allele2 * (allele2 + 1)) / 2) + allele2];
-                            sampleField = StringUtils.join(alleleLikelihoods, ",");
-                        }
-                    }
-                }
+                String sampleField = processSampleField(alleleIdx, formatField, sampleFields[j]);
```

Comment:
honestly, I forgot to do the memory test with the last changes, but I just did it and it seems that the strings are not repeated:

class name | instances
---|---
TreeMap$Entry | 43.4%
TreeMap | 43.4%
Integer | 2.6%
String | 1.5%
char[] | 1.4%

Percentages are over 1.7 GB of heap memory that the program requested